src/commands/login.ts

```diff
@@ -84,10 +84,12 @@ export = command<{ token?: string }>({
             // Generate SAS tokens to check the validity of the input token
             await generateSasTokens();
         } catch (e) {
-            // If token validation fails, revert the saved
-            config.set('iotc.credentials.token', prevInputToken);
-            config.set('iotc.credentials.application', prevApplication);
-            config.set('iotc.credentials.hubs', prevIotHubToken);
+            // If token validation fails, revert to the saved tokens
+            try {
```

Comment:
i probably wouldn't try/catch this. there should be individual checks on each one like this:

```ts
if (prevInputToken) {
    config.set('iotc.credentials.token', prevInputToken);
}
```

reasoning is that if you do a try/catch then you could fail on the first `set()` call and bypass the other two, even if they had valid information to set

Reply:
plus catch-all catch blocks have a way of hiding problems

Reply:
fair enough